modules/mod-ffmpeg/ImportFFmpeg.cpp

```diff
@@ -582,62 +566,68 @@ void FFmpegImportFileHandle::Stop()
 
 void FFmpegImportFileHandle::WriteData(StreamContext *sc, const AVPacketWrapper* packet)
 {
-   // Find the stream index in mStreamContexts array
-   int streamid = -1;
-   auto iter = mChannels.begin();
+   // Find the stream in mStreamContexts array
+   auto streamIt = std::find_if(
+      mStreamContexts.begin(),
+      mStreamContexts.end(),
+      [&](StreamContext& context) { return sc == &context; }
+   );
 
-   for (int i = 0; i < static_cast<int>(mStreamContexts.size()); ++iter, ++i)
-   {
-      if (&mStreamContexts[i] == sc)
-      {
-         streamid = i;
-         break;
-      }
-   }
    // Stream is not found. This should not really happen
-   if (streamid == -1)
+   if (streamIt == mStreamContexts.end())
    {
       //VS: Shouldn't this mean import failure?
       return;
    }
+   auto stream = mStreams[std::distance(mStreamContexts.begin(), streamIt)];
 
-   size_t nChannels = std::min(sc->CodecContext->GetChannels(), sc->InitialChannels);
+   const auto nChannels = std::min(sc->CodecContext->GetChannels(), sc->InitialChannels);
 
+   // Write audio into WaveTracks
    if (sc->SampleFormat == int16Sample)
    {
       auto data = sc->CodecContext->DecodeAudioPacketInt16(packet);
+      const auto channelsCount = sc->CodecContext->GetChannels();
+      const auto samplesPerChannel = data.size() / channelsCount;
 
-      const int channelsCount = sc->CodecContext->GetChannels();
-      const int samplesPerChannel = data.size() / channelsCount;
-
-      // Write audio into WaveTracks
-      auto iter2 = iter->begin();
-      for (size_t chn = 0; chn < nChannels; ++iter2, ++chn)
+      unsigned chn = 0;
+      ImportUtils::ForEachChannel(*stream, [&](auto& channel)
       {
-         iter2->get()->Append(
-            reinterpret_cast<samplePtr>(data.data() + chn), sc->SampleFormat,
+         if(chn >= nChannels)
+            return;
```

Comment:
Why do you make bounds checks on this unsigned index variable only in ImportFFmpeg.cpp but not in the other place that use `ForEachChannel`?  Should there be checks in the other places?

Reply:
I don't think so, I need that to preserve existing loop exit condition.
`const auto nChannels = std::min(sc->CodecContext->GetChannels(), sc->InitialChannels);`
Which formally is not same as `mStreams.size()`, but `nChannels` was used as an upper bound in the original loop